src/distribute_kv_cache/rpc/packet.rs

```diff
@@ -0,0 +1,542 @@
+use std::{collections::HashMap, fmt::Debug, sync::Arc};
+
+use async_trait::async_trait;
+use bytes::{BufMut, BytesMut};
+use tokio::{sync::Mutex, time::Instant};
+use tracing::debug;
+
+use super::{
+    error::RpcError,
+    utils::{get_u64_from_buf, get_u8_from_buf, u64_to_usize},
+};
+
+/// The size of the request header.
+pub const REQ_HEADER_SIZE: u64 = 17;
+/// The size of the response header.
+pub const RESP_HEADER_SIZE: u64 = 17;
+
+/// The Encode trait is used to encode a message structure into a byte buffer.
+pub trait Encode: Sync {
```

Comment:
Have you considered using a encoding lib instead of implementing it by yourself?

Reply:
Encoding lib may be cause extra copy overhead in current scenario, we want to take control of the Buffer.